src/proof/PGHelp.cc

```diff
@@ -220,37 +220,35 @@ bool ProofGraph::mergeClauses(std::vector<Lit> const & A, std::vector<Lit> const
     std::size_t j = 0;
     std::size_t res = 0;
 
+    auto addIfNotPivot = [&resolv, &res, pivot](Lit l) {
+        if (var(l) != pivot) {
+            assert(res == 0 or resolv[res - 1] != l);
+            resolv[res++] = l;
+        }
+    };
+
     while (i < Asize and j < Bsize) {
-        if (A[i] <= B[j]) {
-            if (A[i] == B[j]) { ++j; }
-            if (var(A[i]) != pivot) {
-                assert(res == 0 or resolv[res - 1] != A[i]); // no duplicates
-                resolv[res++] = A[i];
-            }
+        if (A[i] == B[j]) {
+            addIfNotPivot(A[i]);
+            ++i;
+            ++j;
+        } else if (A[i] < B[j]) {
+            addIfNotPivot(A[i]);
```

Comment:
I like the lambda, but here I prefer my version of the `if` structure, so we do not need to call the lambda in both branches.
```
if (A[i] <= B[j] {
	if (A[i] == B[j] { ++j };
	addIfNotPivot(A[i]);
	++i;
}
```

Reply:
Ok, let's go with that then if you prefer!